zerver/lib/avatar.py

```diff
@@ -36,6 +33,39 @@ def avatar_url(
     )
 
 
+def get_system_bots_avatar_file_name(email: str) -> str:
+    system_bot_avatar_name_map = {
+        settings.WELCOME_BOT: "welcome-bot",
+        settings.NOTIFICATION_BOT: "notification-bot",
+        settings.EMAIL_GATEWAY_BOT: "emailgateway",
+    }
+    return urljoin(STATIC_AVATARS_DIR, system_bot_avatar_name_map.get(email, "unknown"))
+
+
+def get_static_avatar_url(email: str, medium: bool) -> str:
+    avatar_file_name = get_system_bots_avatar_file_name(email)
+    avatar_file_name += "-medium.png" if medium else ".png"
+
+    if settings.PRODUCTION and not staticfiles_storage.exists(avatar_file_name):
```

Comment:
If using `settings.PRODUCTION` is somehow problematic here I think checking for `settings.STATIC_ROOT` directly might also be another option,.